project_1/include/server.hpp

```diff
@@ -0,0 +1,46 @@
+#pragma once
+
+#include "packet.hpp"
+
+#include <cstdint>
+#include <vector>
+#include <queue>
+
+// Type for server receiver, contains only ID
+struct ServerBase {
+    ServerBase(std::uint32_t id);
+    ServerBase() = default;
+    virtual ~ServerBase() = default;
+
+    std::uint32_t GetID() const;
+protected:
+    std::uint64_t id;
+};
+
+// Type for sending server, contains ID and distance to receiver
+struct ServerSender : ServerBase {
```

Comment:
In my opinion it would be better to divide `ServerBase` and `ServerSender` into two files